testing/camunda-process-test-java/src/test/java/io/camunda/process/test/impl/mock/BpmnExampleDataReaderTest.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.process.test.impl.mock;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.when;
+
+import io.camunda.client.CamundaClient;
+import io.camunda.client.impl.CamundaObjectMapper;
+import io.camunda.process.test.impl.assertions.util.AwaitilityBehavior;
+import io.camunda.process.test.impl.assertions.util.CamundaAssertJsonMapper;
+import io.camunda.process.test.utils.CamundaAssertExtension;
+import io.camunda.zeebe.model.bpmn.impl.ZeebeConstants;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Answers;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith({CamundaAssertExtension.class, MockitoExtension.class})
+public class BpmnExampleDataReaderTest {
+
+  private static final long PROCESS_DEFINITION_KEY = 1L;
+
+  private BpmnExampleDataReader reader;
+
+  @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+  private CamundaClient client;
+
+  private final CamundaAssertJsonMapper jsonMapper =
+      new CamundaAssertJsonMapper(new CamundaObjectMapper());
+
+  @BeforeEach
+  public void setup() {
+    reader = new BpmnExampleDataReader(client, new AwaitilityBehavior());
+  }
+
+  @Test
+  public void successfulRead() {
+    // given
+    when(client.newProcessDefinitionGetXmlRequest(PROCESS_DEFINITION_KEY).send().join())
+        .thenReturn(bpmnModelWithExampleData());
+
+    // when
+    final Map<String, Object> actual =
+        jsonMapper.fromJsonAsMap(reader.readExampleData(PROCESS_DEFINITION_KEY, "task_send_email"));
+
+    // then
+    final Map<String, Object> expected =
+        Collections.singletonMap(
+            "response", Collections.singletonMap("body", Collections.singletonMap("status", 200)));
+
+    assertThat(actual).isEqualTo(expected);
```

Comment:
🔧 We don't need the JSON mapper to assert the example data. Instead, we could assert the raw JSON string. (same for the test cases below)


```suggestion
    // when
    final String actual = reader.readExampleData(PROCESS_DEFINITION_KEY, "task_send_email");

    // then
    assertThat(actual)
        .isEqualToIgnoringWhitespace("{\"response\":{\"body\":{\"status\":200}}}");
```